### PR TITLE
Migrate "Improve > Payment > Preferences" page

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR migrates Payment preferences page to symfony.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | yes (overrides)
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | You should be able to configure payment modules as you would in old page.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9225)

### CODE REVIEW

File: src/Adapter/Payment/PaymentModulePreferencesConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+namespace PrestaShop\PrestaShop\Adapter\Payment;
+
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Module\Configuration\PaymentRestrictionsConfiguratorInterface;
+use PrestaShop\PrestaShop\Core\Module\DataProvider\PaymentModuleListProviderInterface;
+
+/**
+ * Class PaymentModulePreferencesConfiguration is responsible for configuring payment module restrictions
+ */
+final class PaymentModulePreferencesConfiguration implements DataConfigurationInterface
```
Comment (sarjon) [inline src/Adapter/Payment/PaymentModulePreferencesConfiguration.php]: this class depends on interfaces from `Core` namespace, maybe it should be moved to `Core` namespace as well?
@mickaelandrieu what would be the correct location for it? Maybe `Core/Configuration/Payment`?

**Review by mickaelandrieu — DISMISSED** [review]

**Comment by marionf** [comment]
Hello @sarjon 

When I have no payments modules installed, I have this issue:

capture du 2018-06-28 11-51-41

I should have the following message instead: "No payment module installed"

File: src/PrestaShopBundle/Form/Admin/Improve/Payment/Preferences/PaymentModulePreferencesType.php [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    /**
+     * Sort payment modules by display name
+     *
+     * @param array $paymentModules
+     *
+     * @return array
+     */
+    private function sortPaymentModules(array $paymentModules)
+    {
+        foreach ($paymentModules as $key => $paymentModule) {
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Form/Admin/Improve/Payment/Preferences/PaymentModulePreferencesType.php]: undeclared variable `$sortingBy`

Comment (sarjon) [reply src/PrestaShopBundle/Form/Admin/Improve/Payment/Preferences/PaymentModulePreferencesType.php]: :man_facepalming: that fixes the issue.

**Comment by sarjon** [comment]
@marionf it's fixed now, thanks for quick feedback!

**Comment by marionf** [comment]
Thank you @sarjon 

Sorry, but I just found another little thing :
I have installed Paypal module and the radio buttons are not very pretty 

capture du 2018-06-28 17-14-50

Do you think we can improve that ?

**Comment by sarjon** [comment]
i know, it does not look good to me either. :/ 
Unfortunately, i cannot find any example of how radio should look like in [PrestaKit](https://cdn.rawgit.com/PrestaShop/prestashop-ui-kit/master/index.html). I've also checked Catalog page in Backoffice and it uses this ugly radios as well. :/

maybe @mickaelandrieu you know?

**Comment by mickaelandrieu** [comment]
@sarjon can you ensure we have the right markup here?

https://getbootstrap.com/docs/4.0/components/forms/#default-stacked

Afaik we don't have extra customization that comes from the UI Kit.

**Comment by sarjon** [comment]
> @sarjon can you ensure we have the right markup here?

not exactly the same markup, but as i can see in Boostrap 4 docs, they look the same. 

Edit: the radio markup itself comes from RadioType, i didnt use any custom template here. :)

**Comment by mickaelandrieu** [comment]
Ok, sorry @marionf this is not something we can improve here, but in the PrestaShop UI kit :/

**Comment by marionf** [comment]
All right, I will see with @TristanLDD :)

**Comment by NadiaCamard** [comment]
Hello, for the radio button, can you please use these material icons below? 

`<i class="material-icons">
radio_button_checked
</i>`

`<i class="material-icons">
radio_button_unchecked
</i>`

More info: https://material.io/tools/icons/?style=baseline

**Comment by mickaelandrieu** [comment]
@NadiaCamard sure, but this will be done in another contribution :+1: 

Note the changes you're asking will have an impact on every radio button of the Back Office: are you ok with that?

**Comment by NadiaCamard** [comment]
@mickaelandrieu thanks! that's more than ok if we harmonize the design of this component!

**Comment by mickaelandrieu** [comment]
Hello @sarjon why is it a radio button and not a checkbox input here?

**Comment by sarjon** [comment]
> why is it a radio button and not a checkbox input here?

it depends on `currencies_mode` which is attribute of `PaymentModuleCore`. It can have values `checkbox` or `radio`.

File: src/Adapter/Payment/PaymentModulePreferencesConfiguration.php (hunk 1 above)
Comment (mickaelandrieu) [reply src/Adapter/Payment/PaymentModulePreferencesConfiguration.php]: you're right I do the update

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Thanks @sarjon and @marionf for the review!

**Pull request merged**